osutil/testhelper.go

```diff
@@ -38,3 +38,11 @@ func MustBeTestBinary(panicMsg string) {
 		panic(panicMsg)
 	}
 }
+
+// A checker for the logic that shall not run in a test context
+// ! NOTE: Use this judiciously as it might impose a test-skip risk
+func MustNotBeTestBinary(panicMsg string) {
```

Comment:
I really don't like this as an exported helper. /proc/self/mountinfo it's very annoying but until proven otherwise I would not like many places needing to do this.

Reply:
Hi @pedronis,

Neither am I comfortable with this piece of cod, but fit was already placed within the code already.
All I did was only refactoring to clean it up. 
I also added a very screaming comment for this method in order to deter people from using it.

By the way, I also checked for such things through the code and found few more places that do this,
but for me it didn't stand out. 

So, I would suggest either to get rid of this method from all the places that do this or similar assertion, 
or to use this function as a centralized point to perhaps have better record/tracking of cases that use something like this.

Another thing to take into account, the method on its own doesn't produce test miss.
It's just a helper to warn users/developers that the code that calls to this for some reason cannot be used in the test.
This method is just a warning and in certain way a label that highlights such sensitive points in our code-base.

Kind regards,
Arseniy

Reply:
the code that was there before was not public exported helper, I object to having this as a public exported helper

Reply:
>  several other places

this not true, I think there are exactly two places that needs this pattern (test binary -> panic) here and boot/initramfs.go. It should be reached for rarely. So I still want this function not to be exported and the code to look like previously.

Reply:
@pedronis, I apologize if plural form made you angry. The exact amount wasn't the point. 
What I was trying to say is that the same pattern is used in more then one place, which is true. I only asked, if it will be better to have a centralized place to keep better grouping of them. I understand what you want, I'll do as you say.